### ISSUE TITLE
MMD errror is NAN..

### ISSUE THREAD

**Issue by vinodkkurmi** [HawaiiZeng/mmd-torch#1]
Hi , i am  running this code (without  modifying it) . but getting the .err_mmd_fc7 err: nan	 err_mmd_fc8 err: nan	???
What may be the reason????
Thanks

**Comment by HawaiiZeng** [comment]
Hi vinodkkurmi,
I am relatively busy these days and will come back to this in a few weeks.

**Comment by vinodkkurmi** [comment]
@HawaiiZeng  Sure ..Thanks.......I am  waiting for your reply.. 
One more request , without going deep in the code can you just suggest some possible reason , so that  meanwhile I can  look at them. if you have some time.. thanks..

**Comment by vinodkkurmi** [comment]
@HawaiiZeng  One more thing... Can MMD loss be negative ?????

**Comment by zhFuECL** [comment]
hello there! Has the problem been solved right now?

**Comment by vinodkkurmi** [comment]
@zhFuECL  nan problem has been solved but I am not getting the accuracy as mentioned in the paper.
@HawaiiZeng  Can you help us regarding this issue??

**Comment by zhFuECL** [comment]
@vinodkkurmi @HawaiiZeng I wonder if there is some problem with the backward method. I can't understand why the `criterion:backward(outputs3, self.targets)`  should backward in two stage?

**Comment by zhFuECL** [comment]
@vinodkkurmi  And how do you solve your nan problem? I found I that I am still suffering from this problem

**Comment by HawaiiZeng** [comment]
@vinodkkurmi @zhFuECL  Thank you guys! Sorry I forget most of the implementation details. It has been a long time since I wrote it. Besides, this is written when I just started learning Torch. I would suggest you to track back the code on the author's own website. This repository will be set as private in a few days.(to avoid misguiding others). But still welcome to discuss by email.

**Comment by vinodkkurmi** [comment]
@zhFuECL  there is small bug in the mmd.lua file.  The concat operation of source and target can not be done because dim will not match. So we have to first transpose them then concat them ..To Solve this problem   from line on  68  to  74 line no(mmd.lua file)   replace  this lines.

single_input = single_input:view(self.dim, 1)
        local single_expand = torch.expand(single_input, self.dim, self.total_num)
        local input_concat = torch.cat(source:transpose(1,2), target:transpose(1,2))
        local single_diff2 = torch.pow(single_expand - input_concat, 2) -- dim * total_num
        local sumup = torch.sum(single_diff2, 1) -- the ith row of distance2
